tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Entities/AzurePipelines/Run.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json.Serialization;
+
+namespace Azure.Sdk.Tools.PipelineWitness.Entities.AzurePipelines
+{
+    public class Run
+    {
+        [JsonPropertyName("id")]
+        public string Id => $"{ProjectId}/{PipelineId}/{RunId}";
+
+        [JsonPropertyName("runId")]
+        public int RunId { get; set; }
+
+        [JsonPropertyName("runName")]
+        public string RunName { get; set; }
+
+        [JsonPropertyName("runUrl")]
+        public Uri RunUrl { get; set; }
+
+        [JsonPropertyName("projectId")]
+        public Guid ProjectId { get; set; }
+
+        [JsonPropertyName("projectName")]
+        public string ProjectName { get; set; }
+
+        [JsonPropertyName("projectUrl")]
+        public Uri ProjectUrl { get; set; }
+
+        [JsonPropertyName("pipelineId")]
+        public int PipelineId { get; set; }
+
+        [JsonPropertyName("pipelineName")]
+        public string PipelineName { get; set; }
+
+        [JsonPropertyName("pipelineUrl")]
+        public Uri PipelineUrl { get; set; }
+
+        [JsonPropertyName("repositoryId")]
+        public string RepositoryId { get; set; }
+
+        [JsonPropertyName("reason")]
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        public RunReason Reason { get; set; }
+
+        [JsonPropertyName("result")]
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        public RunResult Result { get; set; }
+
+        [JsonPropertyName("state")]
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        public RunStatus State { get; set; }
+
+        [JsonPropertyName("gitRef")]
+        public string GitReference { get; set; }
+
+        [JsonPropertyName("sha")]
+        public string GitCommitSha { get; set; }
+
+        [JsonPropertyName("startTime")]
+        public DateTimeOffset StartTime { get; set; }
+
+        [JsonPropertyName("finishTime")]
+        public DateTimeOffset FinishTime { get; set; }
+
+        [JsonPropertyName("agentDurationInSeconds")]
```

Comment:
I think worker name might also be interesting. I used it a few times while doing some analysis.

Reply:
Worker name?

Reply:
The timeline records for jobs contain the worker name https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-devops?path=%2Fbuild-queues.ps1&version=GBmaster&line=81&lineEnd=82&lineStartColumn=1&lineEndColumn=1&lineStyle=plain that is how I figured out what was running on an agent and which agent type it was.

Reply:
Ah OK. This record that I'm publishing is one per run, worker name would be per job. When I start extracting the timeline data I can do that for the jobs.